src/main/java/com/exasol/adapter/document/files/FilesSelectionExtractor.java

```diff
@@ -0,0 +1,140 @@
+package com.exasol.adapter.document.files;
+
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import com.exasol.adapter.document.documentfetcher.files.GlobToRegexConverter;
+import com.exasol.adapter.document.mapping.SourceReferenceColumnMapping;
+import com.exasol.adapter.document.queryplanning.selectionextractor.SelectionExtractor;
+import com.exasol.adapter.document.querypredicate.*;
+import com.exasol.adapter.document.querypredicate.normalizer.DnfAnd;
+import com.exasol.adapter.document.querypredicate.normalizer.DnfComparison;
+import com.exasol.adapter.document.querypredicate.normalizer.DnfOr;
+import com.exasol.adapter.sql.SqlLiteralString;
+
+/**
+ * This class applies the predicates on the SOURCE_REFERENCE to the source string and splits the rest of the selection
+ * as post-selection. By that only the required files are read.
+ * <p>
+ * If the selection can not be extracted, (for example because it combines two conditions on SOURCE_REFERENCE using AND
+ * or OR), the source string is not changed, and the whole selection is used as post selection.
```

Comment:
```suggestion
 * If the selection can not be extracted, for example because it combines two conditions on SOURCE_REFERENCE using AND
 * or OR, the source string is not changed, and the whole selection is used as post selection.
```